lightning/src/ln/channel.rs

```diff
@@ -5336,7 +5349,10 @@ impl<SP: Deref> Channel<SP> where
 				(matches!(self.context.channel_state, ChannelState::AwaitingChannelReady(flags) if !flags.is_set(AwaitingChannelReadyFlags::WAITING_FOR_BATCH)) ||
 				matches!(self.context.channel_state, ChannelState::ChannelReady(_)))
 			{
-				self.context.funding_transaction.take()
+				let res = self.context.funding_transaction_unless_rebroadcast();
+				// Note: this is legacy logic, prevents (re)broadcasting twice, unclear if needed
```

Comment:
Not sure what is meant by this. Why is it legacy logic?

Reply:
What I mean by legacy:
- I don't fully understand why it works this way
- I suspect it is not by intention
- but I chose to introduce an extra field just to keep the exact behavior
- changing it would break one unit tests, and who knows what else in the field

Reply:
@TheBlueMatt Any thoughts on making `funding_transaction` a three-state enum instead of introducing a flag and continuing to use `Option`? We'd probably need the flag in the serialization logic, though.

Reply:
Is there any downside for occasional repeated broadcast? At most some unnecessary resource usage?